local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -211,25 +211,31 @@ void WaypointGenerator::smoothWaypoint(float dt) {
 void WaypointGenerator::nextSmoothYaw(float dt) {
   // Use xy smoothing constant for yaw, since this makes more sense than z,
   // and we dont want to introduce yet another parameter
-  const float P_constant_xy = smoothing_speed_xy_;
-  const float D_constant_xy =
-      2.f * std::sqrt(P_constant_xy);  // critically damped
 
-  const float desired_setpoint_yaw = nextYaw(position_, output_.goto_position);
-  const float desired_yaw_velocity = 0.0;
+  if (smoothing_speed_xy_ > 0) {
+    const float P_constant_xy = smoothing_speed_xy_;
+    const float D_constant_xy =
+        2.f * std::sqrt(P_constant_xy);  // critically damped
 
-  float yaw_diff = std::isfinite(desired_setpoint_yaw)
-                       ? desired_setpoint_yaw - setpoint_yaw_
-                       : 0.0f;
+    const float desired_setpoint_yaw_rad =
```

Comment:
I think move this calculation outside of the if, then in the else you can just set it directly.